### ISSUE TITLE
255 character limit on Component->Form->Display Message, error should be handled better

### ISSUE THREAD

**Issue by vesper8** [mautic/mautic#3863]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | X
| Feature request? | 
| Enhancement? | X 

## Description:
When creating a campaign form (Components -> Forms -> New -> Campaign Form
If you set "Successful submit action" to "Display Message"
If you enter more than 255 characters as the Display Message, and click "apply" or "save & close" you get this error under the Name field: 
```
There was an error creating the submission results table. Try a different name.
```

Pretty vague and unhelpful!

If you create the form first, using "test" values for example. And then try to edit it and again add a Display Message of more than 255 characters, this time upon clicking "apply" or "save and close" you get this error:

```
  Uh oh! I think I broke it. If I do it again, please report me to the system administrator! 
500 Internal Server Error
```
Equally vague and unhelpful.

Upon inspection of the mysql table structure for "forms" you see that the post_action_property which holds the "display message" has a varchar 255 limit

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.7.1
| PHP version | 7.0

### Steps to reproduce:
See above
 
### Log errors: 

```
[2017-04-17 08:54:11] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\DriverException: "An exception occurred while executing 'UPDATE forms SET is_published = ?, description = ?, post_action_property = ?, in_kiosk_mode = ?, render_style = ? WHERE id = ?' with params [0, "This campaign is like an information process about what is happening in two weeks.<br \/><br \/>It is all about creating attention, teaching people how this is going to look like in two weeks and starting to create a buzz around Nomad Soulmates. &nbsp;They will also find Information and tips how to present themselves in the webapp, and in the end of two weeks they will be funneled into a big call to action, signing up to the final product.<br \/><br \/>Short:<br \/>- informative:<br \/>*how is the matching working<br \/>*FAQ<br \/>- tips how to present yourself<br \/>- terms and conditions<br \/>- CAT to sign up for the final product", "Yeah! Soon your matching fun begins. Important information such as 'likes' and 'messages' from other users will go straight in your inbox, make sure you open all emails from us to not miss out on a potential date in Bali! Until then we keep you updated and remind you when the matching goes live! Take care, you hear from us soon!", 0, 1, 4]:  SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'post_action_property' at row 1" at /home/forge/mautic.mydomain.com/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 115 {"exception":"[object] (Doctrine\\DBAL\\Exception\\DriverException(code: 0): An exception occurred while executing 'UPDATE forms SET is_published = ?, description = ?, post_action_property = ?, in_kiosk_mode = ?, render_style = ? WHERE id = ?' with params [0, \"This campaign is like an information process about what is happening in two weeks.<br \\/><br \\/>It is all about creating attention, teaching people how this is going to look like in two weeks and starting to create a buzz around Nomad Soulmates. &nbsp;They will also find Information and tips how to present themselves in the webapp, and in the end of two weeks they will be funneled into a big call to action, signing up to the final product.<br \\/><br \\/>Short:<br \\/>- informative:<br \\/>*how is the matching working<br \\/>*FAQ<br \\/>- tips how to present yourself<br \\/>- terms and conditions<br \\/>- CAT to sign up for the final product\", \"Yeah! Soon your matching fun begins. Important information such as 'likes' and 'messages' from other users will go straight in your inbox, make sure you open all emails from us to not miss out on a potential date in Bali! Until then we keep you updated and remind you when the matching goes live! Take care, you hear from us soon!\", 0, 1, 4]:\n\nSQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'post_action_property' at row 1 at /home/forge/mautic.mydomain.com/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:115, Doctrine\\DBAL\\Driver\\PDOException(code: 22001): SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'post_action_property' at row 1 at /home/forge/mautic.mydomain.com/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:93, PDOException(code: 22001): SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'post_action_property' at row 1 at /home/forge/mautic.mydomain.com/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:91)"} []
[2017-04-17 09:01:42] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\DriverException: "An exception occurred while executing 'UPDATE forms SET is_published = ?, post_action_property = ?, in_kiosk_mode = ?, render_style = ? WHERE id = ?' with params [1, "Yeah! Soon your matching fun begins. Important information such as 'likes' and 'messages' from other users will go straight in your inbox, make sure you open all emails from us to not miss out on a potential date in Bali! Until then we keep you updated and remind you when the matching goes live! Take care, you hear from us soon!", 0, 1, 5]:  SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'post_action_property' at row 1" at /home/forge/mautic.mydomain.com/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 115 {"exception":"[object] (Doctrine\\DBAL\\Exception\\DriverException(code: 0): An exception occurred while executing 'UPDATE forms SET is_published = ?, post_action_property = ?, in_kiosk_mode = ?, render_style = ? WHERE id = ?' with params [1, \"Yeah! Soon your matching fun begins. Important information such as 'likes' and 'messages' from other users will go straight in your inbox, make sure you open all emails from us to not miss out on a potential date in Bali! Until then we keep you updated and remind you when the matching goes live! Take care, you hear from us soon!\", 0, 1, 5]:\n\nSQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'post_action_property' at row 1 at /home/forge/mautic.mydomain.com/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:115, Doctrine\\DBAL\\Driver\\PDOException(code: 22001): SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'post_action_property' at row 1 at /home/forge/mautic.mydomain.com/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:93, PDOException(code: 22001): SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'post_action_property' at row 1 at /home/forge/mautic.mydomain.com/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:91)"} []
```

Form validation should be more helpful and mention the character limit. At the very least the form input itself should have
```
<input type="text" maxlength="255">
```

**Comment by ivent** [comment]
Has anyone solved this? 4 months and nothing is still occurring in the latest v2.13.1 version I am days looking for a solution and nothing ????

**Comment by npracht** [comment]
@escopecz you added `WIP` but issue is still existing (just reproduced). Do you have something to share or we simply remove the label?

**Comment by escopecz** [comment]
It's actually resolved by https://github.com/mautic/mautic/pull/7283. Thanks for bringing this up.

**Comment by npracht** [comment]
@vesper8 @ivent can you please test #7283 ?

**Comment by ivent** [comment]
@npracht actually resolved by #7283. Thanks

**Comment by npracht** [comment]
Fixed when #7283 merged.
